### PR TITLE
Fix review decision staying CHANGES_REQUESTED after re-request

### LINKED ISSUE #6: PR still shows as rejected after re-requesting review
## Bug

When a reviewer requests changes on a PR and the author addresses the feedback and re-requests a review, MonitorLizard continues to display the PR with the red ❌ "Changes requested" indicator. The expected behavior is that the review state should reflect the pending re-review (e.g., show as "Review required" or clear the rejection indicator).

## Screenshot

The GitHub UI (left) shows a re-requested review from `gpambrozio`, but MonitorLizard (right) still renders the red rejected icon next to the reviewer name:

Inconsistency

## Root Cause

The app fetches `reviewDecision` from `gh pr view --json reviewDecision`, which is a PR-level summary field. GitHub's API keeps `reviewDecision` as `CHANGES_REQUESTED` even after a re-review is requested — it only updates when the reviewer submits a new review.

To accurately reflect the re-requested state, the app would need to also fetch `reviewRequests` (pending review requests) or `latestReviews` and cross-reference them. If a reviewer appears in both `latestReviews` (with `CHANGES_REQUESTED`) *and* `reviewRequests` (re-requested), the display should show "Review required" instead of "Changes requested."

## Relevant Code

- **Data source**: `GitHubService.fetchPRStatus()` queries `--json reviewDecision` ([GitHubService.swift:343](https://github.com/SeanMcTex/MonitorLizard/blob/main/MonitorLizard/Services/GitHubService.swift#L343))
- **Model**: `ReviewDecision` enum only has `approved`, `changesRequested`, `reviewRequired` ([PullRequest.swift:4-7](https://github.com/SeanMcTex/MonitorLizard/blob/main/MonitorLizard/Models/PullRequest.swift#L4-L7))
- **Display**: `PRRowView` renders the icon based on `pr.reviewDecision` ([PRRowView.swift:134](https://github.com/SeanMcTex/MonitorLizard/blob/main/MonitorLizard/Views/PRRowView.swift#L134))

## Suggested Fix

1. Add `reviewRequests` and/or `latestReviews` to the `gh pr view --json` fields
2. When a reviewer has `CHANGES_REQUESTED` in `latestReviews` but also appears in `reviewRequests`, treat the effective state as `reviewRequired` (pending re-review)
3. This would also enable showing per-reviewer status in the future (e.g., ✅ approved by Alice, ⏳ awaiting Bob's re-review)

### PR DESCRIPTION
**Pull request by SeanMcTex**
## Summary

Fixes #6 (reported by @lukelabonte).

- Fetches `latestReviews` and `reviewRequests` alongside `reviewDecision` in both `fetchPRStatus()` and `fetchOtherPR()`
- Adds `resolveReviewDecision()` helper: when `reviewDecision` is `CHANGES_REQUESTED`, cross-references the two lists — if every reviewer who requested changes is also in the pending re-review list, returns `.reviewRequired` instead
- No change in behavior for approved PRs or PRs with genuinely outstanding change requests

## Root cause

GitHub's `reviewDecision` field stays `CHANGES_REQUESTED` even after the author re-requests a review from the same reviewer. The fix uses `reviewRequests` (pending re-reviews) to detect this case and downgrade the displayed state to "Review required".

## Test plan

- [ ] Open a PR where a reviewer has requested changes, then re-request their review — MonitorLizard should now show "Review required" instead of "Changes requested"
- [ ] Open a PR where a reviewer has requested changes and has *not* been re-requested — should still show "Changes requested"
- [ ] Approved PRs unaffected

### CODE REVIEW

File: MonitorLizard/Services/GitHubService.swift [hunk 1]
```diff
@@ -363,11 +363,42 @@ class GitHubService: ObservableObject {
...
+        let pendingLogins = Set((reviewRequests ?? []).compactMap { $0.login })
+        let changesRequestedLogins = (latestReviews ?? [])
+            .filter { $0.state.uppercased() == "CHANGES_REQUESTED" }
+            .compactMap { $0.author?.login }
+
+        // If every CHANGES_REQUESTED reviewer has a pending re-review request,
+        // the author has addressed the feedback and is awaiting a new review.
+        let allReRequested = !changesRequestedLogins.isEmpty &&
+            changesRequestedLogins.allSatisfy { pendingLogins.contains($0) }
+
```
Comment (lukelabonte) [inline MonitorLizard/Services/GitHubService.swift:407]: This is the bug. When `latestReviews` is empty (which GitHub does return sometimes), `changesRequestedLogins` is empty, so `!changesRequestedLogins.isEmpty` is `false` and we short-circuit to `.changesRequested` even though there's a pending re-review in `reviewRequests`.

```suggestion
        // If every CHANGES_REQUESTED reviewer has a pending re-review request,
        // the author has addressed the feedback and is awaiting a new review.
        if !changesRequestedLogins.isEmpty {
            let allReRequested = changesRequestedLogins.allSatisfy { pendingLogins.contains($0) }
            return allReRequested ? .reviewRequired : .changesRequested
        }

        // GitHub sometimes returns latestReviews without the CHANGES_REQUESTED entry
        // (e.g., older reviews fall off). If the API says CHANGES_REQUESTED but we can't
        // find who requested changes, fall back to checking whether any re-review is pending.
        if !pendingLogins.isEmpty {
            return .reviewRequired
        }

        return .changesRequested
```

File: MonitorLizardTests/PRMonitorViewModelTests.swift [hunk 2]
```diff
@@ -2,6 +2,157 @@ import Testing
...
+    }
+
+    @Test func changesRequestedWithEmptyLatestReviewsStaysChangesRequested() {
+        let result = GitHubService.resolveReviewDecision(
+            rawValue: "CHANGES_REQUESTED",
+            latestReviews: [],
+            reviewRequests: [request("alice")]
+        )
+        #expect(result == .changesRequested)
+    }
```
Comment (lukelabonte) [inline MonitorLizardTests/PRMonitorViewModelTests.swift]: These two tests need updating since the expected behavior changes with the fix. When `latestReviews` is empty/nil but `reviewRequests` has a pending re-review, the result should now be `.reviewRequired`. Also worth adding a test for the truly empty case (no pending requests either).

```suggestion
    @Test func changesRequestedWithNoLatestReviewsAndPendingRequestBecomesReviewRequired() {
        let result = GitHubService.resolveReviewDecision(
            rawValue: "CHANGES_REQUESTED",
            latestReviews: nil,
            reviewRequests: [request("alice")]
        )
        #expect(result == .reviewRequired)
    }

    @Test func changesRequestedWithEmptyLatestReviewsAndPendingRequestBecomesReviewRequired() {
        let result = GitHubService.resolveReviewDecision(
            rawValue: "CHANGES_REQUESTED",
            latestReviews: [],
            reviewRequests: [request("alice")]
        )
        #expect(result == .reviewRequired)
    }

    @Test func changesRequestedWithEmptyLatestReviewsAndNoRequestsStaysChangesRequested() {
        let result = GitHubService.resolveReviewDecision(
            rawValue: "CHANGES_REQUESTED",
            latestReviews: [],
            reviewRequests: []
        )
        #expect(result == .changesRequested)
    }
```

**Review by lukelabonte — CHANGES_REQUESTED** [review]
I pulled this down and tested it, but I'm still seeing the rejected icon on a PR where I've re-requested review.

The issue is that GitHub sometimes returns `latestReviews` as an empty array even when `reviewDecision` is `CHANGES_REQUESTED`. You can repro with:

```
gh pr view 9 --repo gpambrozio/ClaudeCodePlugins --json reviewDecision,latestReviews,reviewRequests
```

That returns `latestReviews: []` but `reviewRequests` has the re-requested reviewer. So `changesRequestedLogins` ends up empty, `allReRequested` short-circuits to `false`, and it stays on `.changesRequested`.

I tested a fix locally and it works. See inline suggestions.

**Review by lukelabonte — APPROVED** [review]
Tested the latest change and everything is working now. The re-requested review state shows correctly. Nice fix!
